File: lib-segment/src/main/java/websiteschema/mpsegment/tools/accurary/ContainErrorAnalyzer.java
package websiteschema.mpsegment.tools.accurary;

import websiteschema.mpsegment.core.WordAtom;

import java.util.Map;

class ContainErrorAnalyzer extends AbstractErrorAnalyzer {

    @Override
    public boolean analysis(WordAtom expect, String possibleErrorWord) {
        boolean foundError = false;
        if (possibleErrorWord.length() == 0) {
            // Bigger word contains the littler words.
            // Should remove the word from dictionary.
            increaseOccur();
//            System.out.println(expect.word + " in " + errorSegment);
            foundError = true;
        } else if (!possibleErrorWord.replaceAll(" ", "").equals(expect.word)
                && possibleErrorWord.contains(expect.word)) {
            increaseOccur();
            addErrorWord(possibleErrorWord);
//            System.out.println(expect.word + " in " + errorSegment);
            foundError = true;
        }
        return foundError;
    }

    @Override
    public void postAnalysis(Map<String, Integer> allWordsAndFreqInCorpus) {
        for(String word : getWords()) {
            if(allWordsAndFreqInCorpus.containsKey(word)) {
                removeErrorWord(word);
            }
        }
    }
}

File: lib-segment-scala/src/main/scala/websiteschema/mpsegment/tools/accurary/ContainErrorAnalyzer.scala
package websiteschema.mpsegment.tools.accurary

import websiteschema.mpsegment.core.Word
import collection.mutable.Map

class ContainErrorAnalyzer extends AbstractErrorAnalyzer {

    override def analysis(expect: Word, possibleErrorWord: String) : Boolean = {
        var foundError = false
        if (possibleErrorWord.length() == 0) {
            // Bigger word contains the littler words.
            // Should remove the word from dictionary.
            increaseOccur()
//            println(expect.word + " in " + possibleErrorWord)
            foundError = true
        } else if (!possibleErrorWord.replaceAll(" ", "").equals(expect.name)
                && possibleErrorWord.contains(expect.name)) {
            increaseOccur()
            addErrorWord(possibleErrorWord)
//            println(expect.word + " in " + possibleErrorWord)
            foundError = true
        }
        return foundError
    }

    override def postAnalysis(allWordsAndFreqInCorpus: Map[String,Int]) {
        for(word <- getWords().keys) {
            if(allWordsAndFreqInCorpus.contains(word)) {
                removeErrorWord(word)
            }
        }
    }
}
